builtin/providers/fastly/resource_fastly_service_v1.go

```diff
@@ -0,0 +1,603 @@
+							Description: "Be strict on checking SSL certs",
+						},
+						// UseSSL is something we want to support in the future, but
+						// requires SSL setup we don't yet have
+						// TODO: Provide all SSL fields from https://docs.fastly.com/api/config#backend
+						// "use_ssl": &schema.Schema{
+						// 	Type:        schema.TypeBool,
+						// 	Optional:    true,
+						// 	Default:     false,
+						// 	Description: "Whether or not to use SSL to reach the Backend",
+						// },
+						"weight": &schema.Schema{
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     100,
+							Description: "How long to wait for the first bytes in milliseconds",
+						},
+					},
+				},
+			},
+
+			"force_destroy": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceServiceV1Create(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+	service, err := conn.CreateService(&gofastly.CreateServiceInput{
+		Name:    d.Get("name").(string),
+		Comment: "Managed by Terraform",
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(service.ID)
+	return resourceServiceV1Update(d, meta)
+}
+
+func resourceServiceV1Update(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	// Update Name. No new verions is required for this
+	if d.HasChange("name") {
+		_, err := conn.UpdateService(&gofastly.UpdateServiceInput{
+			ID:   d.Id(),
+			Name: d.Get("name").(string),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Once activated, Versions are locked and become immutable. This is true for
+	// versions that are no longer active. For Domains, Backends, DefaultHost and
+	// DefaultTTL, a new Version must be created first, and updates posted to that
+	// Version. Loop these attributes and determine if we need to create a new version first
+	var needsChange bool
+	for _, v := range []string{"domain", "backend", "default_host", "default_ttl"} {
+		if d.HasChange(v) {
+			needsChange = true
+		}
+	}
+
+	if needsChange {
+		latestVersion := d.Get("active_version").(string)
+		if latestVersion == "" {
+			// If the service was just created, there is an empty Version 1 available
+			// that is unlocked and can be updated
+			latestVersion = "1"
+		} else {
+			// Clone the latest version, giving us an unlocked version we can modify
+			log.Printf("[DEBUG] Creating clone of version (%s) for updates", latestVersion)
+			newVersion, err := conn.CloneVersion(&gofastly.CloneVersionInput{
+				Service: d.Id(),
+				Version: latestVersion,
+			})
+			if err != nil {
+				return err
+			}
+
+			// The new version number is named "Number", but it's actually a string
+			latestVersion = newVersion.Number
+
+			// New versions are not immediately found in the API, or are not
+			// immediately mutable, so we need to sleep a few and let Fastly ready
+			// itself. Typically, 7 seconds is enough
+			time.Sleep(7 * time.Second)
```

Comment:
Ye olde Cloud Sleep. :cloud: :zzz: :smile: 

We can probably ping Fastly Support to ask about if there's a better heuristic for us to use here. In the meantime, might be good to debug log that we're going to sleep. :+1: